### PR TITLE
Generate all manifests from code

### PR DESCRIPTION
**Pull request by slintes**
**What this PR does / why we need it**:

Generate all manifests from code.
Breaked out as suggested of the operator PR https://github.com/kubevirt/kubevirt/pull/1803

**Release note**:
```release-note
Create all manifests from code, in preparation for the kubevirt operator
```

### CODE REVIEW

**Comment by slintes** [comment]
About make generate: if you set the provider to something not existing, it falls back to the defaults, e.g. `KUBEVIRT_PROVIDER=x make generate`. I didn't have a better idea yet.
Good catch about the j2 templates, I missed that, thanks. Will look into it.
Yes, the log verbosity was the only difference between dev and release manifests. I just forgot about it. Should be possible to get that back.

File: hack/build-manifests.sh [hunk 1]
```diff
@@ -24,7 +24,8 @@ source hack/config.sh
 
 manifest_docker_prefix=${manifest_docker_prefix-${docker_prefix}}
 
-args=$(cd ${KUBEVIRT_DIR}/manifests && find * -type f -name "*.yaml.in")
+# make sure manifests in dev are processed first, so that their result can be used in other manifests (see --dev-manifests-dir)
```
[the remark this answers is not in the document]
Comment (slintes) [reply hack/build-manifests.sh]: The release manifests are using dev manifests now (e.g. `{{index .DevManifests "kubevirt-ns.yaml"}}`). But those manifests are `yaml.in` files and need to be processed first. So `index .DevManifests` uses files from `_out/manifests/dev`. And in order to make sure that the dev manifests are processed first, in order to be available in `_out` for the other manifests, I added the `| sort`. `find` does not guarantee any order.

[the remark this answers is not in the document]
Comment (slintes) [reply hack/build-manifests.sh]: forget my last comment: manifest are created in 2 steps now (first file includes, then variable replacement), dev manifests are taken from the output dir anymore

**Comment by slintes** [comment]
j2 templates are fixed, and that also solved the problem with getting local configuration in the committed manifests :) Will have a look at the verbosity now.
@rmohr do you mean we only need the release manifests, and the dev manifests can be deleted?

**Comment by slintes** [comment]
the only thing which is missing in the j2 templates now is the ImagePullPolicy template. Since this is not a string in the code, it's not possible to get the "{{...}}" value in there...

**Comment by rmohr** [comment]
> the only thing which is missing in the j2 templates now is the ImagePullPolicy template. Since this is not a string in the code, it's not possible to get the "{{...}}" value in there...

Given I understand the comment correctly, `PullPolicy` is just a string in the end. You should have multiple ways in go on how to force-set it to an arbitrary string, or you can load the objects as an "unstructed" k8s object and simply set arbibrary values on the fields.

**Comment by rmohr** [comment]
> @rmohr do you mean we only need the release manifests, and the dev manifests can be deleted?

Pretty much. I found it very useful in the past. I don't think that it is much used by others anymore. That we set a higher verbosity level, is very convenient though.

**Comment by slintes** [comment]
> PullPolicy is just a string in the end

ah yes, sure, thanks for the reminder

File: hack/build-manifests.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (slintes) [reply hack/build-manifests.sh]: and after the last iteration, we no dev manifests at all anymore ;)

File: hack/build-manifests.sh [hunk 2]
```diff
@@ -24,41 +24,59 @@ source hack/config.sh
...
         --container-tag="{{ docker_tag }}" \
         --image-pull-policy="{{ image_pull_policy }}" \
-        --generated-manifests-dir=${KUBEVIRT_DIR}/manifests/generated/ \
-        --input-file=${KUBEVIRT_DIR}/manifests/$arg >${template_outfile}
+        --verbosity=${verbosity} \
+        --input-file=${infile} >${template_outfile}
 done
 
+# Remove tmp files
+(cd ${KUBEVIRT_DIR}/manifests && find . -type f -name "*.yaml.in.tmp" -exec rm {} \;)
```
[the remark this answers is not in the document]
Comment (slintes) [reply hack/build-manifests.sh]: yes, first I ignored them, then they annoyed me too, fixed in the latest commit already ;)

[the remark this answers is not in the document]
Comment (slintes) [reply hack/build-manifests.sh]: oh wait, this is from the latest commit ;)
strange...

**Comment by slintes** [comment]
looks better now. Still some failures especially on os-3.11, but different ones...
@rmohr wdyt?

**Pull request merged**